Automattic/wp-calypso
# App Promo Sidebar: Delete unused/redundant code in App Promo component

A `jetpack/app-branding` feature flag was added in https://github.com/Automattic/wp-calypso/pull/6681 in order to conditionally replace mentions of the WordPress app with the Jetpack app. As the flag is now set to `true` across all environments, we can safely remove all code that mentions the WordPress app, as it is now redundant.

This PR is part of a series that will eventually allow us to remove the `jetpack/app-branding` feature flag.

## Proposed Changes

* This PR iterates on the changes to the Reader's app promo from https://github.com/Automattic/wp-calypso/pull/68293. The now redundant code associated with the `jetpack/app-branding` conditional is removed. 

## Testing Instructions

* Navigate to the Reader in an account where you haven't dismissed the sidebar app promo. (You can manually change `shouldRenderAppPromoSidebar` to `true` [here](https://github.com/Automattic/wp-calypso/blob/4501c55fd4c9394aaf1601ddc94918c5dcd1fb0c/client/reader/sidebar/promo.jsx#L17) if testing locally.)
* Reload the page to cycle through the various app promos to confirm their appearance match with what is displayed in production.

| Desktop promo  | App promo |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/224282718-994c3851-ab64-44d6-b575-d2e3411f0311.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/224125975-43cfa189-41c9-430c-ad8c-e74f4c7018eb.png" width="100%"> |

## Pre-merge Checklist
